### PR TITLE
[GCP] Add billing additional fields

### PR DESCRIPTION
**Pull request by ghost**
## Overview

This change introduces new billing fields and fixes the aggregation functions in the billing dashboard lenses.

Previously, the `max` aggregation was used, since the granularity of the billing data was low. Due to increased granularity which produces multiple documents per day, the `sum` aggregation is appropriate for accurately displaying total costs.

<b>Fields added:</b>
- `sku_id`
- `sku_description`
- `service_id`
- `service_description`
- `effective_price`
- `tags` (nested)
  - `key`
  - `value`

With these additional fields, we can add new lenses to improve the insight into billing data. This should be handled in a separate enhancement PR.

Following the merge of https://github.com/elastic/beats/pull/36902, we can proceed with this PR.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- 

## Screenshots
scrnli_10_31_2023_1-34-56 AM

### CODE REVIEW

**Comment by ishleenk17** [comment]
Lets add the sample data here

**Comment by ishleenk17** [comment]
> Lets add the sample data here

@gpop63 : Please do the needful

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by bhapas — APPROVED** [review]

File: packages/gcp/data_stream/billing/sample_event.json [hunk 1]
```diff
@@ -22,7 +22,13 @@
...
             "project_id": "containerlib-prod-12763",
             "project_name": "elastic-containerlib-prod",
-            "total": 4717.170681
+            "total": 4717.170681,
+            "sku_id": "0D56-2F80-52A5",
+            "service_id": "6F81-5844-456A",
+            "sku_description": "Network Inter Region Ingress from Jakarta to Americas",
+            "service_description": "Compute Engine",
+            "effective_price": 0.00292353,
+            "tags": {}
```
Comment (zmoog) [inline packages/gcp/data_stream/billing/sample_event.json:31]: nit: It would be helpful to include tag values in the sample event.

**Review by zmoog — COMMENTED** [review]
The change looks good; I only have one question and one nit.

**Question**

The [PR](https://github.com/elastic/beats/pull/36902) on Beats introduced a detailed usage cost table alongside the existing standard table. IIRC, users must use the detailed table pattern to get the extra fields.

How do integration users get standard vs. detailed usage cost data? Should they use the detailed table pattern as well?

If users get standard vs. detailed usage cost data by setting the table pattern, we should describe this behavior in the docs and give a hint in the setting description in the UI.

**Nit**

We should include tag values in the sample document.

**Review by zmoog — APPROVED** [review]
Thanks for the docs update!

**Pull request merged**